src/dmd/backend/cdef.d

```diff
@@ -258,6 +258,7 @@ public import dmd.root.longdouble : targ_ldouble = longdouble;
 
 // Extract most significant register from constant
 int REGSIZE();
+extern (C)
```

Comment:
Why this change?

Reply:
On OSX I wasn't able to link the code as cloned:

```
$ make -f posix.mak test
...
Undefined symbols for architecture x86_64:
  "_MSREG", referenced from:
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Error: linker exited with status 1
make[1]: *** [../generated/osx/release/64/dmd] Error 1
```

This change, in a separate commit, seems to fix it.

Would it be better if I split them into two pull requests? This would leave my remaining changes for this pull request not locally tested.

Reply:
It works fine to build for me locally. Please try to clean and do a full rebuild. I'm using DMD 2.085.0 as the host compiler.

Reply:
Thanks, it works now after clean. Removing this commit from this pull request.